### PR TITLE
fix: The image should only be built when explicitly released, or when triggered manually

### PR DESCRIPTION
**Pull request by soulteary**
Currently, every modification to the project will trigger the image build, but in fact it should only trigger the build when a new version is released, or manually trigger the CI build in the background.

Hope it helps.

### CODE REVIEW

**Review by xffxff — COMMENTED** [review]
Thank you for your contribution to the project. You are absolutely right in identifying that the current setup triggers an image build with every modification, which isn't always necessary.

However, I believe a further discussion might be needed. Left some comments

File: .github/workflows/build_docker_image.yml
Comment (soulteary) [reply .github/workflows/build_docker_image.yml:9]: Thanks for the feedback.

In my experience code inspection and build verification are necessary every time CI triggers (if the language needs to be built)

But building the image is unnecessary, because if you don't push it, this process will only cause a waste of time and resources. Regardless of whether insiders are involved.

At present or in the long term, the most frequently updated parts of this warehouse are non-model-related parts, such as documentation and quick start. Perhaps these contents should only be triggered when the maintainer releases a release version?

Want to hear your opinion.

Comment (soulteary) [reply .github/workflows/build_docker_image.yml:7]: > The changes only prevent the build of a docker image upon each pull request. This means that when a pull request is merged, it will still trigger the docker image build.
> 
> Additionally, for pull requests submitted by external contributors, the docker image build process requires explicit authorization from a maintainer. This means it won't trigger automatically for every pull request. You can find the relevant code in https://github.com/01-ai/Yi/blob/main/.github/workflows/build_docker_image.yml#L16-L21

Thanks for asking the question, it's fixed and will only be built when the version is released.

I'd like to submit a new patch in the future about a format review that can be run on every commit.

Just like the points discussed above, in order to make submissions more efficient and high-quality, and reduce unnecessary construction time.

Comment (xffxff) [reply .github/workflows/build_docker_image.yml:7]: > At present or in the long term, the most frequently updated parts of this warehouse are non-model-related parts, such as documentation and quick start. Perhaps these contents should only be triggered when the maintainer releases a release version?

Yes, I agree with you. 

I see a potential benefit in building docker image in pr. This approach allows us to test the docker image created by the specific pull request before it's merged into the main branch. But, as you rightly pointed out, since the most common updates are not code-related, the necessity to update the Docker image for every minor change is indeed debatable. Even when it comes to code changes, we do not need to be so cautious before merging these changes.

@findmyway what do you say?

File: .github/workflows/build_docker_image.yml
Comment (findmyway) [inline .github/workflows/build_docker_image.yml:7]: There's no need to trigger CI on tags. Based on our current policy, tags are created automatically and then docker image is built immediately after that.

Given that most of the existing PRs are docs related, I agree we can switch to manual triggering.

**Review by findmyway — APPROVED** [review]

**Review by xffxff — APPROVED** [review]

**Pull request merged**
